dev/com.ibm.ws.security.javaeesec.cdi/resources/com/ibm/ws/security/javaeesec/cdi/internal/resources/JavaEESecMessages.nlsprops

```diff
@@ -17,50 +17,45 @@
 JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_PARAM.useraction=Make sure that the configuration parameter is set correctly.
 
-JAVAEESEC_CDI_ERROR_PASSWORDHASH_BELOW_MINIMUM_PARAM=CWWKS1934E: The value {0} of the configuration parameter {1} is less than the minimum value {2}.
-JAVAEESEC_CDI_ERROR_PASSWORDHASH_BELOW_MINIMUM_PARAM.explanation=The value of the configuration parameter is less than the minimum value.
+JAVAEESEC_CDI_ERROR_PASSWORDHASH_BELOW_MINIMUM_PARAM=CWWKS1934E: The password hash for the DatabaseIdentityStore attribute is not valid because the {0} value of the {1} configuration parameter is less than the minimum value of {2}.
+JAVAEESEC_CDI_ERROR_PASSWORDHASH_BELOW_MINIMUM_PARAM.explanation=The value of the configuration parameter must be greater than or equal to the minimum value for the password hash of the DatabaseIdentityStore attribute to be valid.
 JAVAEESEC_CDI_ERROR_PASSWORDHASH_BELOW_MINIMUM_PARAM.useraction=Make sure that the configuration parameter value is equal to or greater than the minimum value.
 
-JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA=CWWKS1935E: The hashed data is invalid. {0}
-JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA.explanation=The hashed value in the database is not valid.
-JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA.useraction=Make sure that the hashed data in the database was hashed by the hash algorithm which PasswordHash implementation class supports.
+JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA=CWWKS1935E: The password hash from the database is not valid. {0}
+JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA.explanation=The password hash from the database is not valid, likely because the format of the password is incorrect.  
+JAVAEESEC_CDI_ERROR_PASSWORDHASH_INVALID_DATA.useraction=Make sure that the password hash in the database has the correct format required by the Pbkdf2PasswordHash implementation.
 
-JAVAEESEC_CDI_ERROR_NO_URL=CWWKS1936E: The attribute {0} is not set in webAppSecurity element, the container overrides HttpAuthenticationMechanism will not be functioning properly."
-JAVAEESEC_CDI_ERROR_NO_URL.explanation=Because the required attribute is not set, the container overrides HttpAuthenticationMechanism form authentication is not functioning.
-JAVAEESEC_CDI_ERROR_NO_URL.useraction=the attribute in the message needs to be set.
+JAVAEESEC_CDI_ERROR_NO_URL=CWWKS1936E: The user could not log in. The FormLoginHttpAuthenticationMechanism attribute could not be used for the log in because the {0} attribute is not set in the webAppSecurity element.
+JAVAEESEC_CDI_ERROR_NO_URL.explanation=The user could receive a 401 or 403 error on attempting to log in. The FormLoginHttpAuthenticationMechanism could not be used for the log in because either the loginFormURL attribute or the loginErrorURL attribute is not set in the webAppSecurity element.
+JAVAEESEC_CDI_ERROR_NO_URL.useraction=Make sure that both the loginFormURL attribute and the loginErrorURL attribute are set.
 
```

Comment:
+JAVAEESEC_CDI_ERROR_NO_URL.explanation=The user could receive a 401 or 403 error **when** attempting to log in. The FormLoginHttpAuthenticationMechanism could not be used for the log in because either the loginFormURL attribute or the loginErrorURL attribute is not set in the webAppSecurity element.

Reply:
Fixed.